### ISSUE TITLE
Code cleanup

### ISSUE THREAD

**Issue by ppavlidis** [PavlidisLab/Gemma#154]
It's about time to remove some unused functionality. Some candidates (not definitive or comprehensive, just _candidates_):

- [x] support for terracotta distributed cache (unless we foresee using it...)
- [x] exposure of GO terms in the gene web interface. We still need GO to compute multifunctionality (itself rarely used for anything in Gemma) but showing GO terms on the web interface is out of scope.
- [x] the "expression images" tab on gene pages. This is broken. All underlying code should go.
- [ ] interactive differential expression meta-analysis (DiffExMetaAnalyzerController and related)
- [x] any lingering reference to RServe/Rengine (it is still a dependency for some reason)
- [ ] Phenocarta. This would be a bit painful since we published a paper about it. But we haven't been able to keep it up to date, and either way it doesn't belong in Gemma any more. 
- [x] support for distributed processing within the webapp. This mostly lies under ubic.gemma.core.job.executor. Because the amount of data processing done by the web app has been reduced, this is an unnecessary feature that is difficult to maintain. We still need local background processing, just the distributed option can probably go.
- [ ] support for uploading experiments (from tab-delimited files) directly from the web interface. This is rarely used, since we have not been encouraging users of the system to do this. This functionality is theoretically useful for curators, but we have a CLI that supports upload of TSV

I expect if we remove all of this, in aggregate this will drop >10k lines of code.

**Comment by arteymix** [comment]
For Terracotta, I basically removed its dependencies when I updated EhCache because the client is built-in. The only remaining pieces is the clustered ehcache.xml configuration file and its counterpart in CacheUtils. Removing it completely would be a notable difference because `ehcache.clustered.xml` is 1.7k lines that needs to be kept in sync with `ehcache.nocluster.xml`.

Lately, I started integrating Apache Commons for generating TSV for the new ResultSet endpoint. Notably, we have a standardized CSVFormat object that can be reused and a NumberFormat for floating points representation in https://github.com/PavlidisLab/Gemma/blob/development/gemma-core/src/main/java/ubic/gemma/core/analysis/service/AbstractTsvFileService.java

It can also be used for parsing TSV input.

We can switch from JMS/ActiveMQ to Spring Task. I would love to see this removed because it's blocking me from merging https://github.com/PavlidisLab/Gemma/tree/feature-cli-tools-as-beans. That's because WorkerCLI (which launches a JMS worker) has a completely separate context configuration from the remaining CLI tools.

I managed to remove many lines in the REST API by simplifying how the Arg mechanism is designed (lots of redundant code there) and flattening the service layer.

**Comment by arteymix** [comment]
@ppavlidis Do we actually use the scripts defined under `gemma-core/etc/scripts`? I think they are part of an attempt to do distributed computing with JMS.

**Comment by ppavlidis** [comment]
yes it is part of the JMS framework that is covered by this issue

**Comment by arteymix** [comment]
Terracotta has been fully removed in 08219710d7143ee7de93df9623cba79dd28ac679

**Comment by arteymix** [comment]
Not mentioned here, but we will want to remove `FactorValue.value` at some point. It's been deprecated in https://github.com/PavlidisLab/Gemma/commit/7eb5cb2f3fdfcb88b2b6e5e0d1a9e2a1dd7db2be.

**Comment by ppavlidis** [comment]
I am partially (lazily) implementing a lot of the cleanups described above just by commenting out some of the front-end code

**Comment by arteymix** [comment]
Phenocarta-related classes have been fully deprecated in https://github.com/PavlidisLab/Gemma/commit/93dc64419a192317d1c4dec803e146b8704bcfde.

We don't load the disease ontology anymore as it has been replaced with MONDO, so Phenocarta is effectively broken right now.

**Comment by arteymix** [comment]
When we remove phenocarta, it might be sensible to relocate it's code in a separate module even if we don't make it fully functional. It could be an interesting summer project for a student to revive.

**Comment by ppavlidis** [comment]
Gemma 2 disposition: Won't fix
